libraries/Microsoft.Bot.Schema/ActivityExtensions.cs

```diff
@@ -44,24 +44,26 @@ public static string RemoveRecipientMention(this IMessageActivity activity)
         }
 
         /// <summary>
-        /// Replace any mention text for given id from Text property. First checks for and replaces the name of the
-        /// recipient with the matching id and then checks for the leftover <at></at> tags (this is done to handle
-        /// the way Skype sends mention text.
-        /// Use with caution because this function is altering the text on the Activity.
+        /// Remove any mention text for given id from Text property.  For example, given the message
+        /// @echoBot Hi Bot, this will remove "@echoBot", leaving "Hi Bot".
         /// </summary>
+        /// <description>
+        /// The format of a mention Activity.Entity is dependent on the Channel.  But in all cases we
+        /// expect the Mention.Text to contain the exact text for the user as it appears in
+        /// Activity.Text.
+        /// 
+        /// For example, Teams uses &lt;at&gt;username&lt;/at&gt;, whereas slack use @username. It
+        /// is expected that text is in Activity.Text and this method will remove that value from
+        /// Activity.Text.
+        /// </description>
         /// <param name="activity">activity.</param>
         /// <param name="id">id to match.</param>
         /// <returns>new .Text property value.</returns>
         public static string RemoveMentionText(this IMessageActivity activity, string id)
         {
             foreach (var mention in activity.GetMentions().Where(mention => mention.Mentioned.Id == id))
             {
-                var mentionNameMatch = Regex.Match(mention.Text, @"(?<=<at.*>)(.*?)(?=<\/at>)", RegexOptions.IgnoreCase);
-                if (mentionNameMatch.Success)
-                {
-                    activity.Text = activity.Text.Replace(mentionNameMatch.Value, string.Empty);
-                    activity.Text = Regex.Replace(activity.Text, "<at></at>", string.Empty, RegexOptions.IgnoreCase);
-                }
+                activity.Text = Regex.Replace(activity.Text, mention.Text, string.Empty, RegexOptions.IgnoreCase).Trim();
```

Comment:
The is the original code from the original (pre-May Skype fix commit).  It takes the value of mention.Text (The "text" field of the Entity) and removes it from the Activity.Text.  

Example:
If the Mention Text is "@botname"

and the Activity.Text is "@botname Yo Bot!"

RemoveMentionText will produce "Yo Bot!"

Reply:
Here is the comment above it:

        /// <summary>
        /// Remove any mention text for given id from Text property.  For example, given the message
        /// @echoBot Hi Bot, this will remove "@echoBot", leaving "Hi Bot".
        /// </summary>
        /// <description>
        /// The format of a mention Activity.Entity is dependent on the Channel.  But in all cases we
        /// expect the Mention.Text to contain the exact text for the user as it appears in
        /// Activity.Text.
        /// 
        /// For example, Teams uses &lt;at&gt;username&lt;/at&gt;, whereas slack use @username. It
        /// is expected that text is in Activity.Text and this method will remove that value from
        /// Activity.Text.

Reply:
I commented more on RemoveMentionText and SkypeMentionNormalizeMiddleware, as well as some in the test itself.  All of the the tests follow the same pattern: Mockup an Activity with the channels Entity and test that RemoveMentionText removes it correctly.